### DISCUSSION TITLE
Does next memorise files in public folder?

### DISCUSSION THREAD

**Discussion by Akumzy** [vercel/next.js#19422] [Help] [answered]
I'm using Next with Hapi custom server, now ever since I introduced the Next/Image component I've been having issues with Images showing sometimes, currently, the biggest issue has been once I upload a new image to a folder called uploads in public Next will not see it but once I restart the server it'll be seeing it
![image](https://user-images.githubusercontent.com/25849232/99916331-ebdcbe00-2d09-11eb-8e43-be89ddce26a9.png)
I've traced once which I am very sure the image uploaded but it's not inside the `publicFiles` Set instance

I don't know I have tried to reproduce it with a new project [https://hapi-next-image.herokuapp.com/](https://hapi-next-image.herokuapp.com/) but it's working fine and I don't know what I'm doing wrong in the main project

**Comment by jamesmosier** [comment, marked as answer]
Hi @Akumzy. The `public` folder is only meant to serve static files to be used within your application. It is not meant to be used for dynamic or user uploaded content.

See [the docs](https://nextjs.org/docs/basic-features/static-file-serving) and this note: 

> Note: Only assets that are in the public directory at build time will be served by Next.js. Files added at runtime won't be available. We recommend using a third party service like AWS S3 for persistent file storage.

Therefore for any uploads that happen after your application is built you should use a third-party service like S3 or something similar.

**Reply by Akumzy** [reply]
@jamesmosier Thank you, I've tried to create a dedicated route for serving files through my custom server (Hapi) which seems not to work with Next/Image

**Reply by jamesmosier** [reply]
According to the [src](https://nextjs.org/docs/api-reference/next/image#src) docs, 

> When using an external URL, you must add it to [domains](https://nextjs.org/docs/basic-features/image-optimization#domains) in `next.config.js`.

Therefore because your images are not actually within the Next ecosystem and instead outside of it in your custom server, you would need to treat your images as an external file and add your domain to the `domains` array in your `next.config.js`

**Reply by Akumzy** [reply]
Thanks, I'm on it.
